lib/std/crypto/codecs/base64_hex_ct.zig

```diff
@@ -0,0 +1,463 @@
+            }
+            if (hex_len % 2 != 0) {
+                return error.InvalidPadding;
+            }
+            return hex_len / 2;
+        }
+
+        /// Returns the maximum possible decoded size for a given input length after skipping ignored characters.
+        pub fn decodedLenUpperBound(hex_len: usize) usize {
+            return hex_len / 2;
+        }
+    };
+
+    /// Creates a new decoder that ignores certain characters.
+    /// The decoder will skip any characters that are in the ignore list.
+    /// The ignore list must not contain any valid hexadecimal characters.
+    pub fn decoderWithIgnore(ignore_chars: []const u8) error{InvalidCharacter}!DecoderWithIgnore {
+        var ignored_chars = StaticBitSet(256).initEmpty();
+        for (ignore_chars) |c| {
+            switch (c) {
+                '0'...'9', 'a'...'f', 'A'...'F' => return error.InvalidCharacter,
+                else => if (ignored_chars.isSet(c)) return error.InvalidCharacter,
+            }
+            ignored_chars.set(c);
+        }
+        return DecoderWithIgnore{ .ignored_chars = ignored_chars };
+    }
+
+    fn decodeAny(
+        bin: []u8,
+        hex: []const u8,
+        ignored_chars: ?StaticBitSet(256),
+    ) error{ NoSpaceLeft, InvalidCharacter, InvalidPadding }![]const u8 {
+        var bin_pos: usize = 0;
+        var state: bool = false;
+        var c_acc: u8 = 0;
+        for (hex) |c| {
+            const c_num = c ^ 48;
+            const c_num0: u8 = @truncate((@as(u16, c_num) -% 10) >> 8);
+            const c_alpha: u8 = (c & ~@as(u8, 32)) -% 55;
+            const c_alpha0: u8 = @truncate(((@as(u16, c_alpha) -% 10) ^ (@as(u16, c_alpha) -% 16)) >> 8);
+            if ((c_num0 | c_alpha0) == 0) {
+                if (ignored_chars) |set| {
+                    if (set.isSet(c)) {
+                        continue;
+                    }
+                }
+                return error.InvalidCharacter;
+            }
+            const c_val = (c_num0 & c_num) | (c_alpha0 & c_alpha);
+            if (bin_pos >= bin.len) {
+                return error.NoSpaceLeft;
+            }
+            if (!state) {
+                c_acc = c_val << 4;
+            } else {
+                bin[bin_pos] = c_acc | c_val;
+                bin_pos += 1;
+            }
+            state = !state;
+        }
+        if (state) {
+            return error.InvalidPadding;
+        }
+        return bin[0..bin_pos];
+    }
+};
+
+/// (best-effort) constant time base64 encoding and decoding.
+pub const Base64 = struct {
+    /// The base64 variant to use.
+    pub const Variant = packed struct {
+        /// Use the URL-safe alphabet instead of the standard alphabet.
+        urlsafe_alphabet: bool = false,
+        /// Enable padding with '=' characters.
+        padding: bool = true,
+
+        /// The standard base64 variant.
+        pub const standard = Variant{ .urlsafe_alphabet = false, .padding = true };
```

Comment:
https://github.com/ziglang/zig/issues/5038 is not accepted but `T{}` is definitely discouraged at this point

```suggestion
        pub const standard: Variant = .{ .urlsafe_alphabet = false, .padding = true };
```